File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Functionality/MinimumPropertyPage.vb
Namespace CompanionKit.Controls.ProgressBar.Functionality

	Public Class MinimumPropertyPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        Private Sub mobjIncrementTimer_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjIncrementTimer.Tick
            If ((Me.mobjDemoMin0ProgressBar.Value + Me.mobjDemoMin0ProgressBar.Step) >= Me.mobjDemoMin0ProgressBar.Maximum) Then
                Me.mobjDemoMin0ProgressBar.Value = Me.mobjDemoMin0ProgressBar.Minimum
            End If
            Me.mobjDemoMin0ProgressBar.PerformStep()
            If ((Me.mobjDemoMin75ProgressBar.Value + Me.mobjDemoMin75ProgressBar.Step) >= Me.mobjDemoMin75ProgressBar.Maximum) Then
                Me.mobjDemoMin75ProgressBar.Value = Me.mobjDemoMin75ProgressBar.Minimum
            End If
            Me.mobjDemoMin75ProgressBar.PerformStep()

        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Functionality/MinimumPropertyPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.ProgressBar.Functionality
{
    public partial class MinimumPropertyPage : UserControl
    {
        public MinimumPropertyPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handle Tick event of the timer that increase value of the ProgressBars.
        /// </summary>
        private void mobjIncrementTimer_Tick(object sender, EventArgs e)
        {
            // Validate new value of the the demonstrated ProgressBar pay attention on the set range.
            if (this.mobjDemoMin0ProgressBar.Value + this.mobjDemoMin0ProgressBar.Step >= this.mobjDemoMin0ProgressBar.Maximum)
            {
                this.mobjDemoMin0ProgressBar.Value = this.mobjDemoMin0ProgressBar.Minimum;
            }

            // Increase value of the demonstrated ProgressBar on 
            // a value that is set in the ProgressBar.Step property.
            this.mobjDemoMin0ProgressBar.PerformStep();

            // Compute new value of the the demonstrated ProgressBar
            if (this.mobjDemoMin75ProgressBar.Value + this.mobjDemoMin75ProgressBar.Step >= this.mobjDemoMin75ProgressBar.Maximum)
            {
                this.mobjDemoMin75ProgressBar.Value = this.mobjDemoMin75ProgressBar.Minimum;
            }

            // Increase value of the demonstrated ProgressBar on 
            // a value that is set in the ProgressBar.Step property.
            this.mobjDemoMin75ProgressBar.PerformStep();
        }
    }
}